Integrations/integration-Cymon.yml

```diff
@@ -0,0 +1,543 @@
+                tag = event.get('tag')
+                date = datetime.datetime.strptime(event.get('updated'), '%Y-%m-%dT%H:%M:%SZ')
+
+                if (tag in events and events[tag] < date) or tag not in events:
+                    events.update({tag: date})
+
+            next_link = response.get('next')
+
+        for event in events:
+            events[event] = events[event].strftime("%Y-%m-%d %H:%M:%S")
+
+        return events
+
+    def get_ip_location(ip):
+        url = '{}/{}/{}'.format(SERVER_DASHBOARD_URL_V1, 'geolocation', ip)
+
+        response = http_request('GET', url, DEFAULT_HEADERS)
+
+        lon = response.get('longitude', None)
+        lat = response.get('latitude', None)
+
+        if not lon or not lat:
+            return {}
+        else:
+            return {
+                'lon': lon,
+                'lat': lat
+            }
+
+    def get_ip_domains(ip, max_len):
+        url = '{}/{}/{}/{}?limit={}'.format(SERVER_URL_V1, 'ip', ip, 'domains', max_len)
+        domains = []
+
+        response = http_request('GET', url, DEFAULT_HEADERS)
+
+        for domain in response.get('results', []):
+            date = datetime.datetime.strptime(domain.get('updated'), '%Y-%m-%dT%H:%M:%SZ')
+
+            domains.append({'Hostname': domain.get('name'),
+                            'Last Resolved': date.strftime("%Y-%m-%d %H:%M:%S")})
+
+        return domains
+
+    def get_ip_urls(ip, max_len):
+        url = '{}/{}/{}/{}?limit={}'.format(SERVER_URL_V1, 'ip', ip, 'urls', max_len)
+        urls = {}
+
+        response = http_request('GET', url, DEFAULT_HEADERS)
+
+        for response_url in response.get('results',[]):
+            url = response_url.get('location')
+            if url.endswith("/"):
+                url = url[:-1]
+
+            date = datetime.datetime.strptime(response_url.get('updated'), '%Y-%m-%dT%H:%M:%SZ')
+
+            if (url in urls and urls[url] < date) or url not in urls:
+                urls.update({url: date})
+
+        urls_result = []
+        for url in urls:
+            urls_result.append({'Url': url, "Last Resolved": urls[url].strftime("%Y-%m-%d %H:%M:%S")})
+
+        return urls_result
+
+    def get_ip_asn(ip):
+        url = '{}/{}/{}'.format(SERVER_DASHBOARD_URL_V1, 'ipwhois', ip)
+
+        response = http_request('GET', url, DEFAULT_HEADERS)
+
+        asn = response.get('asn')
+        asn_country_code = response.get('asn_country_code')
+
+        if not asn or not asn_country_code:
+            return {}
+        else:
+            return {
+                'asn': asn,
+                'country': asn_country_code
+            }
+
+    def create_ip_command_markdown(ip, sources, events, domains, urls, asn):
+        md = '## Cymon IP report for: {}\n'.format(ip)
+
+        if asn:
+            md += 'ASN: **{}** ({})\n'.format(asn.get('asn'), asn.get('country'))
+
+        md += '\n'
+
+        if events and sources:
```

Comment:
@MarkShleifer we don't have a case of having `events` **or** `sources`?

Reply:
just wanted to make sure its as fail safe as possible